desktop/src/main/kotlin/io/spine/examples/pingh/desktop/CountdownTimer.kt

```diff
@@ -0,0 +1,128 @@
+import androidx.compose.foundation.layout.size
+import androidx.compose.material3.CircularProgressIndicator
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.produceState
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.graphics.StrokeCap
+import androidx.compose.ui.unit.Dp
+import kotlin.time.Duration.Companion.seconds
+import kotlinx.coroutines.delay
+
+/**
+ * Displays a countdown timer that updates every second, starting from "`minutes`:`seconds`"
+ * and ending at "00:00".
+ *
+ * The timer includes two components:
+ *
+ * 1. A circular progress indicator that visually represents the time left.
+ * 2. A text displaying the exact time remaining in `mm:ss` format.
+ *
+ * The timer starts automatically when the component enters the composition
+ * and stops when it exits.
+ *
+ * @param minutes The number of minutes to count down.
+ * @param seconds The number of seconds to count down.
+ * @param size The timer size.
+ * @param indicatorColor The color of this progress indicator.
+ * @param trackColor The color of the track behind the indicator.
+ */
+@Composable
+internal fun CountdownTimer(
+    minutes: Int,
+    seconds: Int,
+    size: Dp,
+    indicatorColor: Color,
+    trackColor: Color
+) {
+    require(minutes >= 0) { "The number of minutes must be non-negative." }
+    require(seconds >= 0) { "The number of seconds must be non-negative." }
+    val fullTime = remember { TimeUnit(minutes, seconds) }
+    val time by produceState(initialValue = fullTime) {
+        while (value.inWholeSeconds > 0) {
+            delay(1.seconds)
+            value = value.minusSecond()
+        }
+    }
+    Box(
+        modifier = Modifier.size(size),
+        contentAlignment = Alignment.Center
+    ) {
+        CircularProgressIndicator(
+            progress = { time.inWholeSeconds.toFloat() / fullTime.inWholeSeconds },
+            modifier = Modifier.fillMaxSize(),
+            color = indicatorColor,
+            trackColor = trackColor,
+            strokeCap = StrokeCap.Round
+        )
+        Text(
+            text = time.toString(),
+            style = MaterialTheme.typography.bodyLarge
+        )
+    }
+}
+
+/**
+ * A time representation in minutes and seconds for use in countdown timers.
+ */
+private data class TimeUnit(val min: Int, val sec: Int) {
+
+    val inWholeSeconds: Int
+        get() = min * secPerMin + sec
+
+    @Suppress("MagicNumber" /* Uses for ensuring "mm:ss" time format. */)
+    override fun toString(): String {
+        val mins = if (min < 10) "0$min" else "$min"
+        val secs = if (sec < 10) "0$sec" else "$sec"
+        return "$mins:$secs"
+    }
+
+    fun minusSecond(): TimeUnit {
```

Comment:
So basically, for, say, 5 minutes, we will be constantly creating new instances until there's 5 * 60 = 300 of them.

Can we achieve the same somewhat more efficiently?